go/bundle/encoder.go

```diff
@@ -165,6 +165,36 @@ func (is *indexSection) Finalize(ver version.Version) error {
 		if err := enc.EncodeMap(mes); err != nil {
 			return err
 		}
+	} else {
+		// CDDL:
+		// index = {* whatwg-url => [ location-in-responses ] }
+		// whatwg-url = tstr
+		// location-in-responses = (offset: uint, length: uint)
+		m := make(map[string][]*indexEntry)
+		for _, e := range is.es {
+			url := e.URL.String()
+			m[url] = append(m[url], e)
+		}
+
+		mes := []*cbor.MapEntryEncoder{}
+		for url, es := range m {
+			me := cbor.GenerateMapEntry(func(keyE *cbor.Encoder, valueE *cbor.Encoder) {
+				if err := keyE.EncodeTextString(url); err != nil {
+					panic(err)
+				}
+
+				if err := valueE.EncodeUint(es[0].Offset); err != nil {
+					panic(err)
+				}
+				if err := valueE.EncodeUint(es[0].Length); err != nil {
+					panic(err)
+				}
```

Comment:
This generates an invalid CBOR structure, because value of a map is not a single CBOR item.

According to the CDDL:
```
index = {* whatwg-url => [ location-in-responses ] }
location-in-responses = (offset: uint, length: uint)
```
A value of the `index` map is an array of two `uint`s. So you should add
```
valueE.EncodeArrayHeader(2)
```
before encoding the `uint`s.

Reply:
You're right, fixed.